docs/tutorial-01.md

```diff
@@ -0,0 +1,210 @@
+## Building OCI container images
+
+The purpose of this tutorial is to deomonstrate how buildah can be used to build container images compliant with the [Open Container Initiative](https://www.opencontainers.org/) (OCI) [image specification](https://github.com/opencontainers/image-spec). Images can be built from existing images, from scratch, and using Dockerfiles. OCI images built using the buildah command line tool (CLI) and the underlying OCI based technologies (e.g. [containers/image](https://github.com/containers/image) and [containers/storage](https://github.com/containers/storage)) are portable and can therefore run in a docker environment.
+
+In brief the `containers/image` project provides mecahnisms to copy, push, pull, inspect and sign container images. The `containers/storage` project provides mecahnisms for storing filesystem layers, container images, and containers. `buildah` is a CLI that takes advantage of these underlying projects and therefore allows you to build, move, and manage container images and containers.  
+
+First step is to install buildah:
+
+    dnf -y install buildah
+
+After installing buildah we can see there are no images installed. The `buildah images` command will list all the images:
+
+    buildah images
+
+We can also see that there are also no containers by running:
+
+    buildah containers
+  
+When you build a working container from an existing image, buildah defaults to appending '-working-container' to the image's name to construct a name for the container. The buildah CLI conveniently returns the name of the new container. You can take advantage of this by assigning the returned value to a shell varible using standard shell assignment :
+
+    container=$(buildah from fedora)
+
+Of course you can always just run `build from fedora` and then list the containers to see the contianer name. As mentioned previously it will be fedora-working-container in this case.
+
+See the name:  
+
+    echo $container
+
+What can we do with this new container? Let's try running bash:
+
+    buildah run $container bash
+    
+Notice we get a new shell prompt because we running a bash shell inside of the container. It should be noted that `buildah run` is not intended for running production containers. It is for helping debug during the build process. 
+
+Let's try running something else:
+
+    buildah run $container java
+
+Oops. Java is not installed. A message containing something like the following was returned.
+
+    container_linux.go:274: starting container process caused "exec: \"java\": executable file not found in $PATH"
+    
+Lets try installing it using:
+    
+    buildah run $container -- dnf install java
+
+The `--` syntax basically says: no more `buildah run` command options after this point. It is required if the command we specify includes command line options which are not meant for buildah. 
+
+Now running `java` will show that Java has been installed. It will return the `Usage`:
```

Comment:
I think either "\`java\` in the container" or "\`buildah run java\`" would be clearer here.

Reply:
Done. `buildah run $container java`